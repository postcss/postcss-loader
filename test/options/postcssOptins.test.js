import {
  compile,
  getCompiler,
  getErrors,
  getCodeFromBundle,
  getWarnings,
} from '../helpers/index';

describe('Postcss options', () => {
  it('should work "from", "to" and "map" postcssOptions', async () => {
    const compiler = getCompiler('./css/index.js', {
      postcssOptions: {
        from: '/test/from.css',
        to: '/test/to.css',
        map: { inline: false, annotation: false },
      },
      config: false,
    });
    const stats = await compile(compiler);

    const codeFromBundle = getCodeFromBundle('style.css', stats);
    const notNormalizecodeFromBundle = getCodeFromBundle(
      'style.css',
      stats,
      false
    );

    const toIsWork = notNormalizecodeFromBundle.map.file.endsWith('to.css');
    const fromIsWork =
      notNormalizecodeFromBundle.map.sources.filter((i) =>
        i.endsWith('from.css')
      ).length > 0;

    expect(toIsWork).toBe(true);
    expect(fromIsWork).toBe(true);

    expect(codeFromBundle.css).toMatchSnapshot('css');
    expect(codeFromBundle.map).toMatchSnapshot('map');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });
});
