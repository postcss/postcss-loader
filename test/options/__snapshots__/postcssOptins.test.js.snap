// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Postcss options should work "from", "to" and "map" postcssOptions: css 1`] = `
"a { color: black }
"
`;

exports[`Postcss options should work "from", "to" and "map" postcssOptions: errors 1`] = `Array []`;

exports[`Postcss options should work "from", "to" and "map" postcssOptions: map 1`] = `
Object {
  "file": "x",
  "mappings": "AAAA,IAAI,aAAa",
  "names": Array [],
  "sources": Array [
    "xxx",
  ],
  "sourcesContent": Array [
    "a { color: black }
",
  ],
  "version": 3,
}
`;

exports[`Postcss options should work "from", "to" and "map" postcssOptions: warnings 1`] = `Array []`;
